### ISSUE TITLE
Trying to run in docker but I get no output

### ISSUE THREAD

**Issue by sussy-drip** [jcageman/d2mapapi#1]
I've decided to try and run this in a windows based docker container but I am having some trouble, and I wonder if you could help?

Basically I have a local installation of Diablo2 and this is my dockerfile:

```
FROM mcr.microsoft.com/windows:20H2
LABEL maintainer="rmilejcz@gmail.com"
WORKDIR "c:/"
EXPOSE 8080
RUN dism.exe /online /enable-feature /all /featurename:iis-webserver /NoRestart
COPY ["./Diablo2", "c:/Diablo2"]
```

I build it like this:

```
docker build -t d2mapcontainer .
```

and run it like this:

```
docker run -it -p 8080 d2mapcontainer
```

and then I navigate to d2mapapi and run it manually:

```
d2mapapi.exe ../Diablo2
```

On my local machine I am able to run the mapapi fine and I see this output:
```
[2021-10-19 10:48:56.218] TRACE: starting server on 127.0.0.1:8080
[2021-10-19 10:48:56.219]  INFO: init accept #0
[2021-10-19 10:48:56.219]  INFO: server started on 127.0.0.1:8080
```

but when inside the docker container I get NO output

```
C:\Diablo2\mapstuff>d2mapapi.exe ../Diablo2

```

so I'm not sure why it isn't working, but I'm guessing my docker image is missing some dependencies? I am currently working on building `vcpkg` inside Docker to see if installing the dependencies you list in the readme will solve it, but I did not do that on my own system so I am unsure and just wanted to check and see if you had any thoughts as to what could be going wrong

**Comment by sussy-drip** [comment]
Okay so everything is compiling but I'm having trouble with two files:

```
MapRequestHandler.cpp

MapRequestHandler.cpp(30): error C2039: 'scoped_lock': is not a member of 'std'
vcpkg/installed/x86-windows/include\boost/container/detail/std_fwd.hpp(27): note: see declaration of 'std'
MapRequestHandler.cpp(30): error C2065: 'scoped_lock': undeclared identifier
MapRequestHandler.cpp(30): error C2146: syntax error: missing ';' before identifier 'lock'
MapRequestHandler.cpp(30): error C2672: 'lock': no matching overloaded function found
MapRequestHandler.cpp(30): error C2780: 'void std::lock(_Lock0 &,_Lock1 &,_LockN &...)': expects 3 arguments - 1 provided
Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\include\mutex(427): note: see declaration of 'std::lock'


NetworkUtils.cpp

NetworkUtils.cpp(10): error C2556: 'std::string NetworkUtils::hostIP(unsigned short)': overloaded function differs only by return type from 'std::wstring NetworkUtils::hostIP(unsigned short)'
NetworkUtils.h(13): note: see declaration of 'NetworkUtils::hostIP'
NetworkUtils.cpp(10): error C2371: 'NetworkUtils::hostIP': redefinition; different basic types
NetworkUtils.h(13): note: see declaration of 'NetworkUtils::hostIP'
```

any ideas? Thanks again for all your help I really do appreciate it

**Comment by emko** [comment]
https://hub.docker.com/r/mikefallen/d2mapapi
that works i can see the 
[2021-10-19 22:02:00.403] INFO: server started on 127.0.0.1:8080

what i don't get is i can't access it from outside
bridge | 172.17.0.7:8080/TCP | 192.168.1.19:8082

**Comment by scizzydo** [comment]
You need to ensure you're building it with the correct c++ support (17 I believe for scoped_lock)

**Comment by sussy-drip** [comment]
Alright I went ahead and abandoned compiling my own for now, my self compiled exe does not seem to work. I also stopped worrying about windows containers for the time being, using wine was much easier. 

The current release works with this dockerfile (this assumes you have a Diablo2 installation and d2mapapi alongside each other and next to your dockerfile).

```
FROM tianon/wine

ENV WINEARCH=win32
ENV WINEDEBUG=-all

COPY ./Diablo2 /app/game

WORKDIR /app

COPY ./d2mapapi .

EXPOSE 8080

CMD ["wine", "d2mapapi.exe", "/app/game"]
```
